apps/linkedin-to-notion/src/background/index.ts

```diff
@@ -5,12 +5,20 @@ const linkedInURLRegex = /linkedin.com\/in\/.+/;
 chrome.action.onClicked.addListener((tab) => {
   if (tab.url.match(linkedInURLRegex)) {
     chrome.tabs.sendMessage(tab.id, 'toggleLinkedInNotionSidePanel');
+  } else {
+    chrome.tabs.sendMessage(tab.id, 'toggleGoBackToLinkedInSidePanel');
   }
 });
 
 chrome.tabs.onUpdated.addListener((tabId, _, tab) => {
   if (tab.url.match(linkedInURLRegex)) {
     return chrome.tabs.sendMessage(tabId, 'updateLinkedInNotionSidePanel');
   }
-  return chrome.tabs.sendMessage(tabId, 'closeLinkedInNotionSidePanel');
+  return chrome.tabs.sendMessage(tabId, 'closeSidePanels');
+});
+
+chrome.runtime.onMessage.addListener((msg) => {
+  if (msg === 'openLinkedInTab') {
+    chrome.tabs.create({ url: 'https://www.linkedin.com/in/me/' });
+  }
```

Comment:
Let's await the messages from now on
```
chrome.runtime.onMessage.addListener(async (msg) => {
  if (msg === 'openLinkedInTab') {
    return await chrome.tabs.create({ url: 'https://www.linkedin.com/in/me/' });
  }
```

Reply:
Sure but why? I think you know something about that, that I don't.

Reply:
To stay on the main thread. Let's say it fails, by awaiting we know it fails otherwise it's just lost somewhere in pending state.
https://medium.com/technofunnel/javascript-async-await-c83b15950a71